java/test/jmri/jmrix/nce/NceProgrammerManagerTest.java

```diff
@@ -0,0 +1,77 @@
+package jmri.jmrix.nce;
+
+import junit.framework.Assert;
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+import jmri.GlobalProgrammerManager;
+
+/**
+ * JUnit tests for the NceProgrammerManager class
+ *
+ * @author	Bob Jacobsen
+ */
+public class NceProgrammerManagerTest extends TestCase {
+
+    NceSystemConnectionMemo memo;
+
+    public void testDefaultAccess() {
+        // this is checking the "as default ctor built" options, which might not be valid
+        Assert.assertTrue("provides global programmerManager", memo.provides(GlobalProgrammerManager.class));
+        Assert.assertNotNull("global ProgrammerManager exists", memo.get(GlobalProgrammerManager.class));
+        Assert.assertNotNull("global Programmer exists", ((GlobalProgrammerManager)memo.get(GlobalProgrammerManager.class)).getGlobalProgrammer());
```

Comment:
> Why is this cast here?

Because without it, there's a compile error:
````
    [javac] Compiling 1 source file to /Users/jake/Documents/Trains/JMRI/projects/JMRI/java/classes
    [javac] /Users/jake/Documents/Trains/JMRI/projects/JMRI/java/test/jmri/jmrix/nce/NceProgrammerManagerTest.java:41: error: cannot find symbol
    [javac]         Assert.assertNull("no global Programmer exists", memo.get(GlobalProgrammerManager.class).getGlobalProgrammer());
    [javac]                                                                                                 ^
    [javac]   symbol:   method getGlobalProgrammer()
    [javac]   location: class Object
    [javac] 1 error
````